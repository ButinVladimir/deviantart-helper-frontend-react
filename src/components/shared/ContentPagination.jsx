import React from 'react';
import PropTypes from 'prop-types';
import Section from 'react-bulma-components/lib/components/section';
import Container from 'react-bulma-components/lib/components/container';
import Pagination from 'react-bulma-components/lib/components/pagination';

export default function ContentPagination({
  page,
  pageCount,
  showPagination,
  loadPageHandler,
}) {
  return (
    <Section>
      <Container>
        {showPagination && (
          <Pagination
            // Pagination in Bulma starts from 1 while pagination on backend start from 0.
            current={page + 1}
            total={pageCount}
            delta={3}
            onChange={loadPageHandler}
          />
        )}
      </Container>
    </Section>
  );
}

ContentPagination.propTypes = {
  page: PropTypes.number.isRequired,
  pageCount: PropTypes.number.isRequired,
  showPagination: PropTypes.bool.isRequired,
  loadPageHandler: PropTypes.func.isRequired,
};
