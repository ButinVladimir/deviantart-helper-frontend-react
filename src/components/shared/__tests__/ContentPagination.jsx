import React from 'react';
import { shallow } from 'enzyme';
import ContentPagination from '../ContentPagination';

describe('ContentPagination', () => {
  it('can be rendered correctly when pagination is not enabled', () => {
    const wrapper = shallow(<ContentPagination
      page={4}
      pageCount={9}
      showPagination={false}
      loadPageHandler={() => {}}
    />);

    expect(wrapper).toMatchSnapshot();
  });

  it('can be rendered correctly when pagination is enabled', () => {
    const wrapper = shallow(<ContentPagination
      page={4}
      pageCount={9}
      showPagination
      loadPageHandler={() => {}}
    />);

    expect(wrapper).toMatchSnapshot();
  });
});
