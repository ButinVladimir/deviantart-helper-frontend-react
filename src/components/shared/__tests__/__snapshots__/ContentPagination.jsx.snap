// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentPagination can be rendered correctly when pagination is enabled 1`] = `
<Section
  className=""
  renderAs="section"
  size={null}
  style={Object {}}
>
  <Container
    breakpoint={null}
    fluid={false}
    renderAs="div"
    style={Object {}}
  >
    <Pagination
      autoHide={true}
      className=""
      current={5}
      delta={3}
      next="Next"
      onChange={[Function]}
      previous="Previous"
      renderAs="nav"
      showPrevNext={true}
      total={9}
    />
  </Container>
</Section>
`;

exports[`ContentPagination can be rendered correctly when pagination is not enabled 1`] = `
<Section
  className=""
  renderAs="section"
  size={null}
  style={Object {}}
>
  <Container
    breakpoint={null}
    fluid={false}
    renderAs="div"
    style={Object {}}
  />
</Section>
`;
